### PR TITLE
fix:  share replace global define config

### PR DESCRIPTION
**Pull request by IWANABETHATGUY**

### CODE REVIEW

File: crates/rolldown/src/module_loader/module_loader.rs [hunk 1]
```diff
@@ -82,8 +84,29 @@ impl ModuleLoader {
...
-    let common_data =
-      Arc::new(TaskContext { options: Arc::clone(&options), tx, resolver, fs, plugin_driver });
+    let common_data = Arc::new(TaskContext {
+      options: Arc::clone(&options),
+      tx,
+      resolver,
+      fs,
+      plugin_driver,
+      meta: TaskContextMeta {
+        replace_global_define_config: if !options.define.is_empty() {
```
Comment (hyf0) [inline crates/rolldown/src/module_loader/module_loader.rs]: I guess you could just put `ReplaceGlobalDefinesConfig` in the `options.define`.

Comment (IWANABETHATGUY) [reply crates/rolldown/src/module_loader/module_loader.rs]: The only thing I am hesitated to do that is if we want the raw `NormalizedBundlerOptions` in the future. For now the `NormalizedBundlerOptions` is almost `1:1` for `rolldown.config.js`. Is it worth to preserve `1:1` semantic of the original configuration?

Comment (hyf0) [reply crates/rolldown/src/module_loader/module_loader.rs]: No need worried about that. The purpose of `NormalizedBundlerOptions` is make we happy, no need to dealing with something raw. However, the current implementation of the PR is also fine.

**Review by hyf0 — APPROVED** [review]

File: crates/rolldown/src/module_loader/module_loader.rs (hunk 1 above)
Comment (IWANABETHATGUY) [reply crates/rolldown/src/module_loader/module_loader.rs]: Ok, then I will replace it until we find it is necessary to keep the config as is.

Comment (IWANABETHATGUY) [reply crates/rolldown/src/module_loader/module_loader.rs]: If we  replace `define` with `ReplaceGlobalDefinesConfig`(but we need handle error and propagate errors), which will lead change all upstream function return types to `Result<T>`. let's merge it first.

**Pull request merged**